File: 2016/nazionali/nemesi/att/grader.cpp
#include <cstdio>
#include <cassert>
#include <cstdlib>

static FILE *fr, *fw;

// Declaring variables
static int N;
static int* nemico;

// Declaring functions
void smista(int N, int* nemico);

static int gruppi = 0;

void nuovo_gruppo() {
	if (gruppi != 0) fprintf(fw, "\n");
	gruppi++;
}

void aggiungi(int bambino) {
	fprintf(fw, "%d ", bambino);
}

int main() {
	#ifdef EVAL
		fr = fopen("input.txt", "r");
		fw = fopen("output.txt", "w");
	#else
		fr = stdin;
		fw = stdout;
	#endif

	// Reading input
	fscanf(fr, "%d ", &N);
	nemico = (int*)malloc(N * sizeof(int));
	for (int i0 = 0; i0 < N; i0++) {
		fscanf(fr, "%d ", &nemico[i0]);
	}

	// Calling functions
	smista(N, nemico);

	// Writing output
	
	fclose(fr);
	fclose(fw);
	return 0;
}

File: 2016/nazionali/nemesi/att/grader.pas
uses nemesi;

var	
	fr, fw : text;

{ Declaring variables }
	N : Longint;
	nemico : array of Longint;

{ iterators used in for loops }
	i0: LongInt;

begin
{$ifdef EVAL}
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
{$else}
    fr := input;
    fw := output;
{$endif}
    reset(fr);
    rewrite(fw);

	{ Reading input }
	read(fr, N);
	Setlength(nemico, N);
	for i0 := 0 to N-1 do
	begin
		read(fr, nemico[i0]);
	end;

	{ Calling functions }
	smista(N, nemico);

	{ Writing output }
	
	close(fr);
    close(fw);
end.
